lib/internal/modules/esm/get_format.js

```diff
@@ -61,6 +65,8 @@ function defaultGetFormat(url, context, defaultGetFormatUnused) {
           'The Node.js specifier resolution in ESM is experimental.',
           'ExperimentalWarning');
         format = legacyExtensionFormatMap[ext];
+      } else if (statSync(parsed.pathname).isDirectory()) {
```

Comment:
Rather than statting, it would be fine to just check `endsWith('/')` here:
```suggestion
      } else if (parsed.pathname.endsWith('/') {
```

Reply:
This would not catch when users omit the trailing slash though: `import("./some-dir")`

Reply:
It already doesn't though. See the ESM_RESOLVE spec in https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_resolver_algorithm, it special cases `/` for handling `import.meta.resolve('./dir/')` with a trailing slash for easy asset use cases through the same resolver algorithm.